pkg/controllers/infra_interface.go

```diff
@@ -127,6 +128,8 @@ var _ InfraHandler = &FakeInfraHandler{}
 
 type FakeInfraHandler struct{}
 
+type FakeInfraHandlerFailure struct{}
```

Comment:
@zhujian7 a question:
For AWS we have two calls out, infra and IAM. In Azure its only Infra.
To test when Infra returns error, and then when IAM returms error, I will need two separate types. Because
AwsInfraCreator needs to have an err when I want to test it's failure, but I need its error NIL when I test the AwsIamCreator failure.

Any ideas, so I don't have to introduce another type with identical func's except AwsInfraCreator err == nil and AwsIamCreator == errors.New

Reply:
Sorry, I don't have another solution, and I agree with you that it is not worth creating 6x implementations for the new type to test an IAMCreator failure.